### PR TITLE
Update github actions for PRs opened against main

### PR DESCRIPTION
**Pull request by AWNCypher**
- Updates the pr action to include additional checks (linting, security, etc) 
- Updates requirements.txt to include new libs with the pessimistic operator

### CODE REVIEW

**Comment by AWNCypher** [comment]
@cmboling - sorry for the delayed contributions, I will be focusing on this project this week.

**Comment by cmboling** [comment]
LGTM! I'd like to release this with the other PR you have.
Thanks so much for contributing! 🚀 🚀

**Comment by AWNCypher** [comment]
> LGTM! I'd like to release this with the other PR you have as long as the new check passes. Thanks so much for contributing! 🚀 🚀

No problem, I've pushed the fixes for the flake8 findings. Just waiting on the approval to run the action.

**Review by cmboling — APPROVED** [review]

**Pull request merged**
